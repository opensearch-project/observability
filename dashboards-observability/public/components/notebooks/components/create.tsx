/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
/* eslint-disable react-hooks/exhaustive-deps */

import {
    EuiButton,
    EuiFieldText,
    EuiFlexGroup,
    EuiFlexItem,
    EuiForm,
    EuiFormRow,
    EuiHorizontalRule,
    EuiPage,
    EuiPageBody,
    EuiPageContent,
    EuiPageContentHeader,
    EuiPageContentHeaderSection,
    EuiPageHeader,
    EuiPageHeaderSection,
    EuiSpacer,
    EuiTitle,
    EuiToolTip,
  } from '@elastic/eui';
  import DSLService from 'public/services/requests/dsl';
  import React, { ReactChild, useEffect, useState } from 'react';
  import PPLService from 'public/services/requests/ppl';
  import { last } from 'lodash';
  import { AppAnalyticsComponentDeps } from '../home';
  import { PPLReferenceFlyout } from '../../../components/common/helpers';
  import {
    ApplicationRequestType,
  } from '../../../../common/types/application_analytics';
  import { fetchAppById } from '../helpers/utils';
  import "../../../../public/components/application_analytics/app_analytics.scss"
import { ChromeBreadcrumb } from '../../../../../../src/core/public';

  interface CreateNotebookProps {
    // dslService: DSLService;
    // pplService: PPLService;
    existingNotebookId: string;
    parentBreadcrumb: ChromeBreadcrumb;
    setBreadcrumbs: (newBreadcrumbs: ChromeBreadcrumb[]) => void;
  }
  
  export const CreateNotebook = (props: CreateNotebookProps) => {
    const {
        existingNotebookId,
        parentBreadcrumb,
        setBreadcrumbs,
    } = props;
    // const {
    //   parentBreadcrumbs,
    //   chrome,
    //   http,
    //   query,
    //   name,
    //   description,
    //   pplService,
    //   createApp,
    //   updateApp,
    //   setToasts,
    //   setNameWithStorage,
    //   setDescriptionWithStorage,
    //   setQueryWithStorage,
    //   setFilters,
    //   clearStorage,
    //   existingAppId,
    // } = props;
    // const [isFlyoutVisible, setIsFlyoutVisible] = useState(false);
    // const [selectedServices, setSelectedServices] = useState<OptionType[]>([]);
    // const [selectedTraces, setSelectedTraces] = useState<OptionType[]>([]);
  
    const editMode = existingNotebookId !== 'undefined';
    // const [existingApp, setExistingApp] = useState<ApplicationType>({
    //   id: existingAppId,
    //   dateCreated: '',
    //   dateModified: '',
    //   name: '',
    //   description: '',
    //   baseQuery: '',
    //   servicesEntities: [],
    //   traceGroups: [],
    //   panelId: '',
    //   availability: { name: '', color: '', availabilityVisId: '' },
    // });
  
    useEffect(() => {
      setBreadcrumbs([
        parentBreadcrumb,
        {
          text: 'Notebooks',
          href: '#/notebooks',
        },
        {
          text: editMode ? 'Edit' : 'Create',
          href: `#/notebooks/${editMode ? 'edit' : 'create'}`,
        },
      ]);
    }, []);
  
    // useEffect(() => {
    //   if (editMode && existingNotebookId) {
    //     fetchAppById(
    //       http,
    //       pplService,
    //       existingAppId,
    //       setExistingApp,
    //       setFilters,
    //       () => {},
    //       setToasts
    //     );
    //   }
    // }, [existingNotebookId]);
  
    // useEffect(() => {
    //   if (editMode) {
    //     setNameWithStorage(existingApp.name);
    //     setDescriptionWithStorage(existingApp.description);
    //     setQueryWithStorage(existingApp.baseQuery);
    //   }
    // }, [existingApp]);
  
    // const closeFlyout = () => {
    //   setIsFlyoutVisible(false);
    // };
  
    // let flyout;
    // if (isFlyoutVisible) {
    //   flyout = <PPLReferenceFlyout module="explorer" closeFlyout={closeFlyout} />;
    // }
  
    // const isDisabled = !name || (!query && !selectedTraces.length && !selectedServices.length);
  
    // const missingField = (needLog: boolean) => {
    //   let popoverContent = '';
    //   if (isDisabled || (needLog && !query)) {
    //     if (!name) {
    //       popoverContent = 'Name is required.';
    //     } else if (!query && !selectedServices.length && !selectedTraces.length) {
    //       popoverContent = 'Provide at least one log source, service, entity or trace group.';
    //     } else if (needLog && !query) {
    //       popoverContent = 'Log source is required to set availability.';
    //     }
    //     return <p>{popoverContent}</p>;
    //   }
    // };
  
    // const onCreate = (type: string) => {
    //   const appData = {
    //     name,
    //     description,
    //     baseQuery: query,
    //     servicesEntities: selectedServices.map((option) => option.label),
    //     traceGroups: selectedTraces.map((option) => option.label),
    //     panelId: '',
    //     availabilityVisId: '',
    //   };
    //   createApp(appData, type);
    // };
  
    // const onUpdate = () => {
    //   const appData = {
    //     name,
    //     description,
    //     servicesEntities: selectedServices.map((option) => option.label),
    //     traceGroups: selectedTraces.map((option) => option.label),
    //   };
    //   updateApp(existingAppId, appData, 'update');
    // };
  
    // const onCancel = () => {
    //   clearStorage();
    //   window.location.assign(`${last(parentBreadcrumbs)!.href}application_analytics`);
    // };
  
    return (
      <div style={{ maxWidth: '1130px' }}>
        <EuiPage /*style={{flexDirection: 'column'}}*/>
          <EuiPageBody component="div">
            <EuiPageHeader>
              <EuiPageHeaderSection>
                <EuiTitle data-test-subj="createPageTitle" size="l">
                    <h1>{editMode ? 'Edit' : 'Create'} Notebook</h1>
                </EuiTitle>
              </EuiPageHeaderSection>
            </EuiPageHeader>
            
            
            <EuiPageContent id="notebookInfo">
              <EuiPageContentHeader>
                <EuiPageContentHeaderSection>
                  <EuiTitle size="m">
                    <h2>Notebook information</h2>
                  </EuiTitle>
                </EuiPageContentHeaderSection>
              </EuiPageContentHeader>
              <EuiHorizontalRule />
               <EuiForm component="form">
                <EuiFormRow label="Name" data-test-subj="nameFormRow">
                  <EuiFieldText
                    name="name"
                    value={"RANDOM FOR NOW"}
                    // onChange={(e) => setNameWithStorage(e.target.value)}
                  />
                </EuiFormRow>
              </EuiForm>
            </EuiPageContent>
            </EuiPageBody>
            {/* </div> */}
            {/* <EuiSpacer />
            <EuiPageContent id="composition">
              <EuiPageContentHeader>
                <EuiPageContentHeaderSection>
                  <EuiTitle size="m">
                    <h2>Composition</h2>
                  </EuiTitle>
                </EuiPageContentHeaderSection>
              </EuiPageContentHeader>
              <EuiHorizontalRule />
              <LogConfig editMode={editMode} setIsFlyoutVisible={setIsFlyoutVisible} {...props} />
              <EuiHorizontalRule />
              <ServiceConfig
                selectedServices={selectedServices}
                setSelectedServices={setSelectedServices}
                {...props}
              />
              <EuiHorizontalRule />
              <TraceConfig
                selectedTraces={selectedTraces}
                setSelectedTraces={setSelectedTraces}
                {...props}
              />
            </EuiPageContent>
            <EuiSpacer />
            <EuiFlexGroup>
              <EuiFlexItem grow={false}>
                <EuiButton data-test-subj="cancelCreateButton" onClick={onCancel}>
                  Cancel
                </EuiButton>
              </EuiFlexItem>
              <EuiFlexItem grow={false}>
                <EuiToolTip position="top" content={missingField(false)}>
                  <EuiButton
                    data-test-subj="createButton"
                    isDisabled={isDisabled}
                    onClick={editMode ? onUpdate : () => onCreate('create')}
                    fill={editMode ? true : false}
                  >
                    {editMode ? 'Save' : 'Create'}
                  </EuiButton>
                </EuiToolTip>
              </EuiFlexItem>
              {editMode || (
                <EuiFlexItem grow={false}>
                  <EuiToolTip position="top" content={missingField(true)}>
                    <EuiButton
                      data-test-subj="createAndSetButton"
                      fill
                      isDisabled={isDisabled || !query}
                      onClick={() => onCreate('createSetAvailability')}
                    >
                      Create and Set Availability
                    </EuiButton>
                  </EuiToolTip>
                </EuiFlexItem>
              )} */} 
            {/* </EuiFlexGroup>
          </EuiPageBody> */}
              </EuiPage>
      </div>
    );
  };
  