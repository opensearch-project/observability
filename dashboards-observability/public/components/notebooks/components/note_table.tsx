/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import {
  EuiButton,
  EuiContextMenuItem,
  EuiContextMenuPanel,
  EuiFieldSearch,
  EuiFlexGroup,
  EuiFlexItem,
  EuiHorizontalRule,
  EuiInMemoryTable,
  EuiLink,
  EuiOverlayMask,
  EuiPage,
  EuiPageBody,
  EuiPageContent,
  EuiPageContentHeader,
  EuiPageContentHeaderSection,
  EuiPageHeader,
  EuiPageHeaderSection,
  EuiPopover,
  EuiSpacer,
  EuiTableFieldDataColumnType,
  EuiText,
  EuiTitle,
} from '@elastic/eui';
import CSS from 'csstype';
import _ from 'lodash';
import moment from 'moment';
import React, { ReactElement, useEffect, useState } from 'react';
import { ChromeBreadcrumb } from '../../../../../../src/core/public';
import {
  CREATE_NOTE_MESSAGE,
  NOTEBOOKS_DOCUMENTATION_URL,
} from '../../../../common/constants/notebooks';
import { UI_DATE_FORMAT } from '../../../../common/constants/shared';
import {
  DeleteNotebookModal,
  getCustomModal,
  getSampleNotebooksModal,
} from './helpers/modal_containers';
import { NotebookType } from './main';

const pageStyles: CSS.Properties = {
  float: 'left',
  width: '100%',
  maxWidth: '1130px',
};

type NoteTableProps = {
  loading: boolean;
  fetchNotebooks: () => void;
  addSampleNotebooks: () => void;
  notebooks: Array<NotebookType>;
  createNotebook: (newNoteName: string) => void;
  renameNotebook: (newNoteName: string, noteId: string) => void;
  cloneNotebook: (newNoteName: string, noteId: string) => void;
  deleteNotebook: (noteId: string, noteName?: string, showToast?: boolean) => void;
  parentBreadcrumb: ChromeBreadcrumb;
  setBreadcrumbs: (newBreadcrumbs: ChromeBreadcrumb[]) => void;
  setToast: (title: string, color?: string, text?: string) => void;
};

export function NoteTable(props: NoteTableProps) {
  const [isModalVisible, setIsModalVisible] = useState(false); // Modal Toggle
  const [modalLayout, setModalLayout] = useState(<EuiOverlayMask></EuiOverlayMask>); // Modal Layout
  const [isActionsPopoverOpen, setIsActionsPopoverOpen] = useState(false);
  const [selectedNotebooks, setSelectedNotebooks] = useState([]);
  const [searchQuery, setSearchQuery] = useState('');
  const { notebooks, createNotebook, renameNotebook, cloneNotebook, deleteNotebook } = props;

  useEffect(() => {
    props.setBreadcrumbs([
      props.parentBreadcrumb,
      {
        text: 'Notebooks',
        href: '#/notebooks',
      },
    ]);
    props.fetchNotebooks();
  }, []);

  const closeModal = () => {
    setIsModalVisible(false);
  };
  const showModal = () => {
    setIsModalVisible(true);
  };

  const onCreate = async (newNoteName: string) => {
    createNotebook(newNoteName);
    closeModal();
  };

  const onRename = async (newNoteName: string) => {
    renameNotebook(newNoteName, selectedNotebooks[0].id);
    closeModal();
  };

  const onClone = async (newName: string) => {
    cloneNotebook(newName, selectedNotebooks[0].id);
    closeModal();
  };

  const onDelete = async () => {
    const toastMessage = `Notebook${
      selectedNotebooks.length > 1 ? 's' : ' ' + selectedNotebooks[0].path
    } successfully deleted!`;
    Promise.all(selectedNotebooks.map((notebook) => deleteNotebook(notebook.id, undefined, false)))
      .then(() => props.setToast(toastMessage))
      .catch((err) => {
        props.setToast(
          'Error deleting notebooks, please make sure you have the correct permission.',
          'danger'
        );
        console.error(err.body.message);
      });
    closeModal();
  };

  const createNote = () => {
    setModalLayout(
      getCustomModal(
        onCreate,
        closeModal,
        'Name',
        'Create notebook',
        'Cancel',
        'Create',
        undefined,
        CREATE_NOTE_MESSAGE
      )
    );
    showModal();
  };

  const renameNote = () => {
    setModalLayout(
      getCustomModal(
        onRename,
        closeModal,
        'Name',
        'Rename notebook',
        'Cancel',
        'Rename',
        selectedNotebooks[0].path,
        CREATE_NOTE_MESSAGE
      )
    );
    showModal();
  };

  const cloneNote = () => {
    setModalLayout(
      getCustomModal(
        onClone,
        closeModal,
        'Name',
        'Duplicate notebook',
        'Cancel',
        'Duplicate',
        selectedNotebooks[0].path + ' (copy)',
        CREATE_NOTE_MESSAGE
      )
    );
    showModal();
  };

  const deleteNote = () => {
    const notebookString = `notebook${selectedNotebooks.length > 1 ? 's' : ''}`;
    setModalLayout(
      <DeleteNotebookModal
        onConfirm={onDelete}
        onCancel={closeModal}
        title={`Delete ${selectedNotebooks.length} ${notebookString}`}
        message={`Are you sure you want to delete the selected ${selectedNotebooks.length} ${notebookString}?`}
      />
    );
    showModal();
  };

  const addSampleNotebooks = async () => {
    setModalLayout(
      getSampleNotebooksModal(closeModal, async () => {
        closeModal();
        await props.addSampleNotebooks();
      })
    );
    showModal();
  };

  const popoverButton = (
    <EuiButton
      iconType="arrowDown"
      iconSide="right"
      onClick={() => setIsActionsPopoverOpen(!isActionsPopoverOpen)}
    >
      Actions
    </EuiButton>
  );

  const popoverItems: ReactElement[] = [
    <EuiContextMenuItem
      key="rename"
      disabled={notebooks.length === 0 || selectedNotebooks.length !== 1}
      onClick={() => {
        setIsActionsPopoverOpen(false);
        renameNote();
      }}
    >
      Rename
    </EuiContextMenuItem>,
    <EuiContextMenuItem
      key="duplicate"
      disabled={notebooks.length === 0 || selectedNotebooks.length !== 1}
      onClick={() => {
        setIsActionsPopoverOpen(false);
        cloneNote();
      }}
    >
      Duplicate
    </EuiContextMenuItem>,
    <EuiContextMenuItem
      key="delete"
      disabled={notebooks.length === 0 || selectedNotebooks.length === 0}
      onClick={() => {
        setIsActionsPopoverOpen(false);
        deleteNote();
      }}
    >
      Delete
    </EuiContextMenuItem>,
    <EuiContextMenuItem
      key="addSample"
      onClick={() => {
        setIsActionsPopoverOpen(false);
        addSampleNotebooks();
      }}
    >
      Add samples
    </EuiContextMenuItem>,
  ];

  const tableColumns = [
    {
      field: 'path',
      name: 'Name',
      sortable: true,
      truncateText: true,
      render: (value, record) => (
        <EuiLink href={`#/notebooks/${record.id}`}>{_.truncate(value, { length: 100 })}</EuiLink>
      ),
    },
    {
      field: 'dateModified',
      name: 'Last updated',
      sortable: true,
      render: (value) => moment(value).format(UI_DATE_FORMAT),
    },
    {
      field: 'dateCreated',
      name: 'Created',
      sortable: true,
      render: (value) => moment(value).format(UI_DATE_FORMAT),
    },
  ] as Array<
    EuiTableFieldDataColumnType<{
      path: string;
      id: string;
      dateCreated: string;
      dateModified: string;
    }>
  >;

  return (
    <div style={pageStyles}>
      <EuiPage>
        <EuiPageBody component="div">
          <EuiPageHeader>
            <EuiPageHeaderSection>
              <EuiTitle size="l">
                <h1>Notebooks</h1>
              </EuiTitle>
            </EuiPageHeaderSection>
          </EuiPageHeader>
          <EuiPageContent id="notebookArea">
            <EuiPageContentHeader>
              <EuiPageContentHeaderSection>
                <EuiTitle size="s">
                  <h3>
                    Notebooks<span className="panel-header-count"> ({notebooks.length})</span>
                  </h3>
                </EuiTitle>
                <EuiSpacer size="s" />
                <EuiText size="s" color="subdued">
                  Use Notebooks to interactively and collaboratively develop rich reports backed by
                  live data. Common use cases for notebooks includes creating postmortem reports,
                  designing run books, building live infrastructure reports, or even documentation.{' '}
                  <EuiLink external={true} href={NOTEBOOKS_DOCUMENTATION_URL} target="blank">
                    Learn more
                  </EuiLink>
                </EuiText>
              </EuiPageContentHeaderSection>
              <EuiPageContentHeaderSection>
                <EuiFlexGroup gutterSize="s">
                  <EuiFlexItem>
                    <EuiPopover
                      panelPaddingSize="none"
                      button={popoverButton}
                      isOpen={isActionsPopoverOpen}
                      closePopover={() => setIsActionsPopoverOpen(false)}
                    >
                      <EuiContextMenuPanel items={popoverItems} />
                    </EuiPopover>
                  </EuiFlexItem>
                  <EuiFlexItem>
                    <EuiButton fill onClick={() => createNote()}>
                      Create notebook
                    </EuiButton>
                  </EuiFlexItem>
                </EuiFlexGroup>
              </EuiPageContentHeaderSection>
            </EuiPageContentHeader>
            <EuiHorizontalRule margin="m" />
            {notebooks.length > 0 ? (
              <>
                <EuiFieldSearch
                  fullWidth
                  placeholder="Search notebook name"
                  value={searchQuery}
                  onChange={(e) => setSearchQuery(e.target.value)}
                />
                <EuiHorizontalRule margin="m" />
                <EuiInMemoryTable
                  loading={props.loading}
                  items={
                    searchQuery
                      ? notebooks.filter((notebook) =>
                          notebook.path.toLowerCase().includes(searchQuery.toLowerCase())
                        )
                      : notebooks
                  }
                  itemId="id"
                  columns={tableColumns}
                  tableLayout="auto"
                  pagination={{
                    initialPageSize: 10,
                    pageSizeOptions: [8, 10, 13],
                  }}
                  sorting={{
                    sort: {
                      field: 'dateModified',
                      direction: 'desc',
                    },
                  }}
                  allowNeutralSort={false}
                  isSelectable={true}
                  selection={{
                    onSelectionChange: (items) => setSelectedNotebooks(items),
                  }}
                />
              </>
            ) : (
              <>
                <EuiSpacer size="xxl" />
                <EuiText textAlign="center">
                  <h2>No notebooks</h2>
                  <EuiSpacer size="m" />
                  <EuiText color="subdued">
                    Use notebooks to create post-mortem reports, build live infrastructure
                    <br />
                    reports, or foster explorative collaborations with data.
                  </EuiText>
                </EuiText>
                <EuiSpacer size="m" />
                <EuiFlexGroup justifyContent="center">
                  <EuiFlexItem grow={false}>
                    <EuiButton fullWidth={false} onClick={() => createNote()}>
                      Create notebook
                    </EuiButton>
                  </EuiFlexItem>
                  <EuiFlexItem grow={false}>
                    <EuiButton fullWidth={false} onClick={() => addSampleNotebooks()}>
                      Add samples
                    </EuiButton>
                  </EuiFlexItem>
                </EuiFlexGroup>
                <EuiSpacer size="xxl" />
              </>
            )}
          </EuiPageContent>
        </EuiPageBody>
      </EuiPage>
      {isModalVisible && modalLayout}
    </div>
  );
}
