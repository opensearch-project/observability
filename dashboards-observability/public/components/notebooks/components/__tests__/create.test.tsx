import { fireEvent, render } from '@testing-library/react';
import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { CreateNotebook } from '../create';
import { NoteTable } from '../note_table';

describe('<CreateNotebook /> spec', () => {
  configure({ adapter: new Adapter() });

  it('renders the edit notebook component', () => {
    const createNotebook = jest.fn();
    const renameNotebook = jest.fn();
    const setBreadcrumbs = jest.fn();
    const utils = render(
      <CreateNotebook
        existingNotebookId={decodeURIComponent("random")}
        parentBreadcrumb={{ href: 'parent-href', text: 'parent-text' }}
        setBreadcrumbs={setBreadcrumbs}
        createNotebook={createNotebook}
        renameNotebook={renameNotebook}
      />
    );
    expect(utils.container.firstChild).toMatchSnapshot();
  });

  it('renders the create notebook component', () => {
    const createNotebook = jest.fn();
    const renameNotebook = jest.fn();
    const setBreadcrumbs = jest.fn();
    const utils = render(
      <CreateNotebook
        existingNotebookId={encodeURIComponent('')}
        parentBreadcrumb={{ href: 'parent-href', text: 'parent-text' }}
        setBreadcrumbs={setBreadcrumbs}
        createNotebook={createNotebook}
        renameNotebook={renameNotebook}
      />
    );
    expect(utils.container.firstChild).toMatchSnapshot();
  });
});
