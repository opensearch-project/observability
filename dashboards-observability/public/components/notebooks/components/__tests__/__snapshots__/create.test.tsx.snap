// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateNotebook /> spec renders the create notebook component 1`] = `
<div
  style="max-width: 1130px;"
>
  <div
    class="euiPage euiPage--paddingMedium euiPage--grow"
  >
    <div
      class="euiPageBody euiPageBody--borderRadiusNone"
    >
      <header
        class="euiPageHeader euiPageHeader--responsive euiPageHeader--center"
      >
        <div
          class="euiPageHeaderSection"
        >
          <h1
            class="euiTitle euiTitle--large"
            data-test-subj="createPageTitle"
          >
            Edit
             Notebook
          </h1>
        </div>
      </header>
      <div
        class="euiPanel euiPanel--paddingLarge euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiPageContent"
        id="notebookInfo"
        role="main"
      >
        <div
          class="euiPageContentHeader euiPageContentHeader--responsive"
        >
          <div
            class="euiPageContentHeaderSection"
          >
            <h2
              class="euiTitle euiTitle--medium"
            >
              Notebook information
            </h2>
          </div>
        </div>
        <hr
          class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginLarge"
        />
        <form
          class="euiForm"
        >
          <div
            class="euiFormRow"
            data-test-subj="nameFormRow"
            id="random_html_id-row"
          >
            <div
              class="euiFormRow__labelWrapper"
            >
              <label
                class="euiFormLabel euiFormRow__label"
                for="random_html_id"
              >
                Name
              </label>
            </div>
            <div
              class="euiFormRow__fieldWrapper"
            >
              <div
                class="euiFormControlLayout"
              >
                <div
                  class="euiFormControlLayout__childrenWrapper"
                >
                  <input
                    class="euiFieldText"
                    id="random_html_id"
                    name="name"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
      <div
        class="euiSpacer euiSpacer--l"
      />
      <div
        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <button
            class="euiButton euiButton--primary"
            data-test-subj="cancelCreateButton"
            type="button"
          >
            <span
              class="euiButtonContent euiButton__content"
            >
              <span
                class="euiButton__text"
              >
                Cancel
              </span>
            </span>
          </button>
        </div>
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <span
            class="euiToolTipAnchor"
          >
            <button
              class="euiButton euiButton--primary euiButton--fill euiButton-isDisabled"
              data-test-subj="createButton"
              disabled=""
              type="button"
            >
              <span
                class="euiButtonContent euiButton__content"
              >
                <span
                  class="euiButton__text"
                >
                  Save
                </span>
              </span>
            </button>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<CreateNotebook /> spec renders the edit notebook component 1`] = `
<div
  style="max-width: 1130px;"
>
  <div
    class="euiPage euiPage--paddingMedium euiPage--grow"
  >
    <div
      class="euiPageBody euiPageBody--borderRadiusNone"
    >
      <header
        class="euiPageHeader euiPageHeader--responsive euiPageHeader--center"
      >
        <div
          class="euiPageHeaderSection"
        >
          <h1
            class="euiTitle euiTitle--large"
            data-test-subj="createPageTitle"
          >
            Edit
             Notebook
          </h1>
        </div>
      </header>
      <div
        class="euiPanel euiPanel--paddingLarge euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiPageContent"
        id="notebookInfo"
        role="main"
      >
        <div
          class="euiPageContentHeader euiPageContentHeader--responsive"
        >
          <div
            class="euiPageContentHeaderSection"
          >
            <h2
              class="euiTitle euiTitle--medium"
            >
              Notebook information
            </h2>
          </div>
        </div>
        <hr
          class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginLarge"
        />
        <form
          class="euiForm"
        >
          <div
            class="euiFormRow"
            data-test-subj="nameFormRow"
            id="random_html_id-row"
          >
            <div
              class="euiFormRow__labelWrapper"
            >
              <label
                class="euiFormLabel euiFormRow__label"
                for="random_html_id"
              >
                Name
              </label>
            </div>
            <div
              class="euiFormRow__fieldWrapper"
            >
              <div
                class="euiFormControlLayout"
              >
                <div
                  class="euiFormControlLayout__childrenWrapper"
                >
                  <input
                    class="euiFieldText"
                    id="random_html_id"
                    name="name"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
      <div
        class="euiSpacer euiSpacer--l"
      />
      <div
        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <button
            class="euiButton euiButton--primary"
            data-test-subj="cancelCreateButton"
            type="button"
          >
            <span
              class="euiButtonContent euiButton__content"
            >
              <span
                class="euiButton__text"
              >
                Cancel
              </span>
            </span>
          </button>
        </div>
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <span
            class="euiToolTipAnchor"
          >
            <button
              class="euiButton euiButton--primary euiButton--fill euiButton-isDisabled"
              data-test-subj="createButton"
              disabled=""
              type="button"
            >
              <span
                class="euiButtonContent euiButton__content"
              >
                <span
                  class="euiButton__text"
                >
                  Save
                </span>
              </span>
            </button>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
